src/renderer/src/pages/settings/settings.tsx

```diff
@@ -27,32 +27,31 @@ export function Settings() {
     window.electron.updateUserPreferences(values);
   };
 
-  const renderCategory = () => {
-    if (currentCategory === "general") {
-      return (
-        <SettingsGeneral
-          userPreferences={userPreferences}
-          updateUserPreferences={handleUpdateUserPreferences}
-        />
-      );
+  function renderCategory() {
+    switch (currentCategory) {
+      case "general":
+        return (
+          <SettingsGeneral
+            userPreferences={userPreferences}
+            updateUserPreferences={handleUpdateUserPreferences}
+          />
+        );
+      case "real_debrid":
+        return (
+          <SettingsRealDebrid
+            userPreferences={userPreferences}
+            updateUserPreferences={handleUpdateUserPreferences}
+          />
+        );
+      default:
```

Comment:
If you want to refactor to a switch case (which makes sense to me) I think it would be better to use the "behaviour" string here instead of default

Reply:
Can I return the "return" of the case statement btw without having to write return for every case? Tried to do that but does not work, i am more familiar with ruby that has explicit return

Reply:
AFAIK, in javasript you have to return on each case